### DISCUSSION TITLE
How do i set/input a mask from another program?

### DISCUSSION THREAD

**Discussion by intangir** [lllyasviel/Fooocus#1123] [Q&A]
I want to input an image and a mask instead of selecting the mask in the GUI. the gui masking is a little clumsy,  if i could do a png mask outside and drag it in that would be handy sometimes, or have a transparent png and have it fill in the transparent regions?

**Comment by intangir** [comment]
i guess its called input mask uploading in automatic1111, basically i want to make a mask in gimp or the like, and use that (or reuse the same mask on multiple inputs), can fine tune the mask alot more deliberately in a real editor

**Comment by xhoxye** [comment]
Maybe you can try this https://github.com/lllyasviel/Fooocus/pull/1126

**Reply by intangir** [reply]
That seems to work great! thanks, im still playing around with it now

**Reply by intangir** [reply]
well it worked pretty decently on a large person shaped masked in figure

but it doesn't work right on small masked out text, (for text removal over complex backgrounds)

it leaves a grayish blur

**Reply by xhoxye** [reply]
Small masks are best applied by hand
